OrderBulk - insert in closed session

It happens sometimes in checkpoint block
```
DEBUG[03-17|04:03:10] TomoX Lending State Root                 number=29700 parent=0x2d2970eb895c0337bc526dcc23ee668b36882af43154e3c017ae642181aaa8ee nextRoot=0x2d2970eb895c0337bc526dcc23ee668b36882af43154e3c017ae642181aaa8ee
DEBUG[03-17|04:03:10] logExchangeData takes                    time=132.715ms    blockNumber=29699
panic: Session already closed
goroutine 1045383 [running]:
github.com/globalsign/mgo.(*Session).cluster(...)
    /root/pkg/mod/github.com/globalsign/mgo@v0.0.0-20181015135952-eeefdecb41b8/session.go:2071
github.com/globalsign/mgo.(*Session).acquireSocket(0xc026d87860, 0xc02b22eb00, 0x0, 0x0, 0x0)
    /root/pkg/mod/github.com/globalsign/mgo@v0.0.0-20181015135952-eeefdecb41b8/session.go:5141 +0x51e
github.com/globalsign/mgo.(*Collection).writeOp(0xc01d52f290, 0x102e6c0, 0xc02463d4d0, 0x1, 0x0, 0x0, 0x0)
    /root/pkg/mod/github.com/globalsign/mgo@v0.0.0-20181015135952-eeefdecb41b8/session.go:5312 +0x8e
github.com/globalsign/mgo.(*Bulk).runInsert(0xc01d52f2c0, 0xc079cf6d40, 0xc02b0af540, 0xc01eadf040, 0x1)
    /root/pkg/mod/github.com/globalsign/mgo@v0.0.0-20181015135952-eeefdecb41b8/bulk.go:325 +0xb6
github.com/globalsign/mgo.(*Bulk).Run(0xc01d52f2c0, 0xc02af1fd20, 0x0, 0x0)
    /root/pkg/mod/github.com/globalsign/mgo@v0.0.0-20181015135952-eeefdecb41b8/bulk.go:295 +0x252
github.com/tomochain/tomochain/tomoxDAO.(*MongoDatabase).CommitBulk(0xc0000fe0e0, 0x126abce, 0x13)
    /root/go/src/github.com/tomochain/tomochain/tomoxDAO/mongodb.go:327 +0x62
github.com/tomochain/tomochain/tomox.(*TomoX).SyncDataToSDKNode(0xc00004a400, 0xc06f694c80, 0xc71b9fab00421ad, 0xd967410bbb814404, 0x9c26690fd6866725, 0xfc169a2bfcd3b081, 0x1b98d1c0, 0xed60242fe, 0x0, 0xc072cd6d20, ...)
    /root/go/src/github.com/tomochain/tomochain/tomox/tomox.go:499 +0x4389
github.com/tomochain/tomochain/core.(*BlockChain).logExchangeData(0xc0001a1000, 0xc0036bf9e0)
    /root/go/src/github.com/tomochain/tomochain/core/blockchain.go:2534 +0x868
github.com/tomochain/tomochain/core.(*BlockChain).insertBlock(0xc0001a1000, 0xc0036bf9e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
    /root/go/src/github.com/tomochain/tomochain/core/blockchain.go:1981 +0x1a99
github.com/tomochain/tomochain/core.(*BlockChain).InsertBlock(0xc0001a1000, 0xc0036bf9e0, 0x7ba04c3d445f09e4, 0x0)
    /root/go/src/github.com/tomochain/tomochain/core/blockchain.go:1678 +0x39
github.com/tomochain/tomochain/eth.NewProtocolManager.func6(0xc0036bf9e0, 0xc00034cdf0, 0xc0036bf9e0)
    /root/go/src/github.com/tomochain/tomochain/eth/handler.go:208 +0x342
github.com/tomochain/tomochain/eth/fetcher.(*Fetcher).insert.func1(0xc0002c8ff0, 0xe800b69f796c9383, 0xd919b103745e1c6b, 0xc9406a88003d6588, 0x7ba04c3d445f09e4, 0xc072208068, 0xc002a03aa0, 0x10)
    /root/go/src/github.com/tomochain/tomochain/eth/fetcher/fetcher.go:713 +0x1566
created by github.com/tomochain/tomochain/eth/fetcher.(*Fetcher).insert
    /root/go/src/github.com/tomochain/tomochain/eth/fetcher/fetcher.go:655 +0x2f3
INFO [03-17|04:03:11] Maximum peer count                       ETH=25 LES=0 total=25
DEBUG[03-17|04:03:11] FS scan times                            list=69.945µs set=5.132µs diff=3.08µs
INFO [03-17|04:03:11] Allocated cache and file handles         database=/root/scripts/nodes/7/tomox cache=128 handles=1024
```

--- a/tomox/order_processor.go
+++ b/tomox/order_processor.go
@@ -715,8 +715,7 @@ func (tomox *TomoX) UpdateMediumPriceBeforeEpoch(epochNumber uint64, tradingStat
 // orderbook hash genereted from baseToken, quoteToken at tomochain/tomox/tradingstate/common.go:214
 func (tomox *TomoX) LogEpochPrice(epochNumber uint64, epochPriceResult map[common.Hash]*big.Int) error {
 	db := tomox.GetMongoDB()
-	sc := db.InitBulk()
-	defer sc.Close()
+	db.InitBulk()
 
 	for orderbook, price := range epochPriceResult {
 		if price.Sign() <= 0 {

--- a/tomox/tomox.go
+++ b/tomox/tomox.go
@@ -277,8 +277,7 @@ func (tomox *TomoX) SyncDataToSDKNode(takerOrderInTx *tradingstate.OrderItem, tx
 		err                                 error
 	)
 	db := tomox.GetMongoDB()
-	sc := db.InitBulk()
-	defer sc.Close()
+	db.InitBulk()
 	// 1. put processed takerOrderInTx to db
 	lastState := tradingstate.OrderHistoryItem{}
 	val, err := db.GetObject(takerOrderInTx.Hash, &tradingstate.OrderItem{})
@@ -550,8 +549,7 @@ func (tomox *TomoX) UpdateOrderCache(baseToken, quoteToken common.Address, order
 
 func (tomox *TomoX) RollbackReorgTxMatch(txhash common.Hash) error {
 	db := tomox.GetMongoDB()
-	sc := db.InitBulk()
-	defer sc.Close()
+	db.InitBulk()
 
 	items := db.GetListItemByTxHash(txhash, &tradingstate.OrderItem{})
 	if items != nil {

--- a/tomoxDAO/interfaces.go
+++ b/tomoxDAO/interfaces.go
@@ -5,7 +5,6 @@
 package tomoxDAO
 
 import (
-	"github.com/globalsign/mgo"
 	"github.com/tomochain/tomochain/common"
 	"github.com/tomochain/tomochain/ethdb"
 )
@@ -27,11 +26,11 @@ type TomoXDAO interface {
 	DeleteItemByTxHash(txhash common.Hash, val interface{})
 
 		// basic tomox
-		InitBulk() *mgo.Session
+		InitBulk()
 		CommitBulk() error
 
 		// tomox lending
-		InitLendingBulk() *mgo.Session
+		InitLendingBulk()
 		CommitLendingBulk() error
 
 	// leveldb methods

--- a/tomoxDAO/leveldb.go
+++ b/tomoxDAO/leveldb.go
@@ -3,7 +3,6 @@ package tomoxDAO
 import (
 	"bytes"
 	"encoding/hex"
-	"github.com/globalsign/mgo"
 	"github.com/tomochain/tomochain/common"
 	"sync"
 
@@ -118,16 +117,14 @@ func (db *BatchDatabase) GetListItemByHashes(hashes []string, val interface{}) i
 	return []interface{}{}
 }
 
-func (db *BatchDatabase) InitBulk() *mgo.Session {
-	return nil
+func (db *BatchDatabase) InitBulk() {
 }
 
 func (db *BatchDatabase) CommitBulk() error {
 	return nil
 }
 
-func (db *BatchDatabase) InitLendingBulk() *mgo.Session {
-	return nil
+func (db *BatchDatabase) InitLendingBulk() {
 }
 
 func (db *BatchDatabase) CommitLendingBulk() error {

--- a/tomoxDAO/mongodb.go
+++ b/tomoxDAO/mongodb.go
@@ -303,21 +303,19 @@ func (db *MongoDatabase) DeleteObject(hash common.Hash, val interface{}) error {
 	return nil
 }
 
-func (db *MongoDatabase) InitBulk() *mgo.Session {
-	sc := db.Session.Copy()
+func (db *MongoDatabase) InitBulk() {
+	sc := db.Session
 	db.orderBulk = sc.DB(db.dbName).C(ordersCollection).Bulk()
 	db.tradeBulk = sc.DB(db.dbName).C(tradesCollection).Bulk()
 	db.epochPriceBulk = sc.DB(db.dbName).C(epochPriceCollection).Bulk()
-	return sc
 }
 
-func (db *MongoDatabase) InitLendingBulk() *mgo.Session {
-	sc := db.Session.Copy()
+func (db *MongoDatabase) InitLendingBulk() {
+	sc := db.Session
 	db.lendingItemBulk = sc.DB(db.dbName).C(lendingItemsCollection).Bulk()
 	db.lendingTradeBulk = sc.DB(db.dbName).C(lendingTradesCollection).Bulk()
 	db.topUpBulk = sc.DB(db.dbName).C(lendingTopUpCollection).Bulk()
 	db.repayBulk = sc.DB(db.dbName).C(lendingRepayCollection).Bulk()
-	return sc
 }
 
 func (db *MongoDatabase) CommitBulk() error {

--- a/tomoxlending/tomoxlending.go
+++ b/tomoxlending/tomoxlending.go
@@ -208,8 +208,7 @@ func (l *Lending) SyncDataToSDKNode(takerLendingItem *lendingstate.LendingItem, 
 		err                                             error
 	)
 	db := l.GetMongoDB()
-	sc := db.InitLendingBulk()
-	defer sc.Close()
+	db.InitLendingBulk()
 	// 1. put processed takerLendingItem to database
 	lastState := lendingstate.LendingItemHistoryItem{}
 	// Typically, takerItem has never existed in database
@@ -424,8 +423,8 @@ func (l *Lending) SyncDataToSDKNode(takerLendingItem *lendingstate.LendingItem, 
 
 func (l *Lending) UpdateLiquidatedTrade(result lendingstate.FinalizedResult, trades map[common.Hash]*lendingstate.LendingTrade) error {
 	db := l.GetMongoDB()
-	sc := db.InitLendingBulk()
-	defer sc.Close()
+	db.InitLendingBulk()
+
 
 	txhash := result.TxHash
 	txTime := time.Unix(0, (result.Timestamp/1e6)*1e6).UTC() // round to milliseconds
@@ -622,8 +621,7 @@ func (l *Lending) UpdateLendingTradeCache(hash common.Hash, txhash common.Hash, 
 
 func (l *Lending) RollbackLendingData(txhash common.Hash) error {
 	db := l.GetMongoDB()
-	sc := db.InitLendingBulk()
-	defer sc.Close()
+	db.InitLendingBulk()
 
 	// rollback lendingItem
 	items := db.GetListItemByTxHash(txhash, &lendingstate.LendingItem{})

